ocsp/responder/redis/checked_redis_source_test.go

```diff
@@ -0,0 +1,184 @@
+package redis
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/go-gorp/gorp/v3"
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/db"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/metrics"
+	"github.com/letsencrypt/boulder/ocsp/responder"
+	ocsp_test "github.com/letsencrypt/boulder/ocsp/test"
+	"github.com/letsencrypt/boulder/test"
+	"golang.org/x/crypto/ocsp"
+)
+
+// echoSource implements rocspSourceInterface, returning the provided response
+// and panicking if signAndSave is called.
+type echoSource struct {
+	resp *ocsp.Response
+}
+
+func (es echoSource) Response(ctx context.Context, req *ocsp.Request) (*responder.Response, error) {
+	return &responder.Response{Response: es.resp, Raw: es.resp.Raw}, nil
+}
+
+func (es echoSource) signAndSave(ctx context.Context, req *ocsp.Request, cause string) (*responder.Response, error) {
+	panic("should not happen")
+}
```

Comment:
One option would be for `echoSource` to embed an `errorSource`, just like `recordingEchoSource` embeds an `echoSource`, to prevent having to reimplement this method on both fakes.

The same is true for `errorSelector` and `echoSelector` below.